src/work-allocation-2/utils/work-allocation-utils.ts

```diff
@@ -127,23 +128,45 @@ export const servicesMap: {[key: string]: string} =  {
   SCSS: 'Social security and child support'
 };
 
-export function getOptions(taskRoles: TaskRole[]): OptionsModel[] {
+export function getOptions(taskRoles: TaskRole[], sessionStorageService: ISessionStorageService): OptionsModel[] {
   const options = new Array<OptionsModel>();
+  // Consider role categories only with either OWN or EXECUTE permissions
   const roleCategories = taskRoles.filter(role => role.role_category !== null && role.role_category !== undefined
     && (roleIncludes(role.permissions, Permissions.Own) || roleIncludes(role.permissions, Permissions.Execute))).
     map(taskRole => taskRole.role_category as RoleCategory);
+
+  // Decide the category to be selected by default
+  const roleCategoryToSelectByDefault = getRoleCategoryToBeSelectedByDefault(taskRoles, sessionStorageService);
   roleCategories.forEach(roleCategory => {
     if (!options.find(option => option.optionId === roleCategory)) {
-      options.push({
-          optionId: roleCategory,
-          optionValue: roleCategory,
-          label: this.getLabel(roleCategory)
-        });
+      const option: OptionsModel = {
+        optionId: roleCategory,
+        optionValue: roleCategory,
+        label: this.getLabel(roleCategory)
+      };
+      if (roleCategory === roleCategoryToSelectByDefault) {
+        option.checked = 'checked';
+      }
+      options.push(option);
     }
   });
   return options;
 }
 
+export function getRoleCategoryToBeSelectedByDefault(taskRoles: TaskRole[], sessionStorageService: ISessionStorageService): RoleCategory {
+  // Consider only role categories with OWN permission for radio button default selection
+  const uniqueRoleCategoriesWithOwnPermissions = taskRoles.filter(role => role.role_category !== null && role.role_category !== undefined
```

Comment:
Could we not just say 'role.roleCategory &&' here instead of the null and undefined checks?